### PR TITLE
upgrade-tiller changes

### PR DESCRIPTION
**Pull request by kopiczko**
Note: this is change proposal to #71.

### CODE REVIEW

File: helmclient.go [hunk 1]
```diff
@@ -815,23 +816,31 @@ func getTillerImage(pod *corev1.Pod) (string, error) {
...
-		return false, microerror.Mask(err)
+		return microerror.Mask(err)
 	}
 
-	currentTillerVersion, err := parseTillerVersion(currentTillerImage)
+	currentVersion, err := parseTillerVersion(currentImage)
 	if err != nil {
-		return false, microerror.Mask(err)
+		return microerror.Mask(err)
 	}
```
Comment (kopiczko) [inline helmclient.go]: I noticed that below we were using `TillerImageSpec` const which we shouldn't as we specify `tillerImage` field now. This probably breaks some tests.

File: integration/test/upgradetiller/upgrade_tiller_test.go [hunk 2]
```diff
@@ -19,55 +19,64 @@ func TestUpgradeTiller(t *testing.T) {
 
 	var err error
 
-	labelSelector := "app=helm,name=tiller"
+	currentTillerImage := "quay.io/giantswarm/tiller:v2.8.2"
 	outdatedTillerImage := "gcr.io/kubernetes-helm:v2.7.2"
+
+	labelSelector := "app=helm,name=tiller"
 	tillerNamespace := "giantswarm"
```
Comment (kopiczko) [inline integration/test/upgradetiller/upgrade_tiller_test.go]: Below `currentHelmClient` and `outdatedHelmClient` are set and used in the test.

File: helmclient.go (hunk 1 above)
Comment (kopiczko) [reply helmclient.go]: Unit tests fixed.

File: integration/test/upgradetiller/upgrade_tiller_test.go [hunk 3]
```diff
@@ -19,55 +19,64 @@ func TestUpgradeTiller(t *testing.T) {
...
 
-	var helmClient *helmclient.Client
+	var currentHelmClient *helmclient.Client
+	{
+		currentHelmClient = config.HelmClient
+	}
+
+	var outdatedHelmClient *helmclient.Client
```
Comment (rossf7) [inline integration/test/upgradetiller/upgrade_tiller_test.go]: This is nice and yes I agree makes the test easier to read.

File: helmclient.go [hunk 4]
```diff
@@ -815,23 +816,31 @@ func getTillerImage(pod *corev1.Pod) (string, error) {
 	return tillerImage, nil
 }
 
-func isTillerOutdated(pod *corev1.Pod) (bool, error) {
-	currentTillerImage, err := getTillerImage(pod)
+func validateTillerVersion(pod *corev1.Pod, desiredImage string) error {
```
Comment (rossf7) [inline helmclient.go]: Yes changing the method signature back makes sense. As it gets called from newTunnel.

File: spec.go [hunk 5]
```diff
@@ -7,14 +7,12 @@ import (
...
-	// TillerImageSpec is the tiller image to be installed.
-	TillerImageSpec = "quay.io/giantswarm/tiller:v2.8.2"
-
 	// defaultMaxHistory is the maximum number of release versions stored per
 	// release by default.
 	defaultMaxHistory = 10
 
-	tillerDefaultNamespace = "kube-system"
+	defaultTillerImage     = "quay.io/giantswarm/tiller:v2.8.2"
+	defaultTillerNamespace = "kube-system"
```
Comment (rossf7) [inline spec.go]: <3

**Comment by rossf7** [comment]
> Feel free to merge it to your branch if you find it ok. We can also work on #73 before merging.

Sorry I missed this. Going to merge and will continue with testing.

**Pull request merged**